internal/daemon/controller/controller.go

```diff
@@ -49,6 +49,11 @@ type downstreamRouter interface {
 	// StartRouteMgmtTicking starts a ticker which manages the router's
 	// connections.
 	StartRouteMgmtTicking(context.Context, func() string, int) error
+
+	// ProcessPendingConnections starts a function that continually processes
+	// incoming client connections and processes them. This never returns until
```

Comment:
Nit: this reads a bit weird 😁.
```suggestion

	// ProcessPendingConnections starts a function that continually processes
	// incoming client connections. This never returns until
```